### PR TITLE
Fix docs

### PR DESCRIPTION
**Pull request by roblabla**
WIP, don't merge yet.

### CODE REVIEW

File: libuser/src/vi.rs [hunk 1]
```diff
@@ -21,6 +25,10 @@ impl ViInterface {
         }
     }
 
+    /// Create a new window of the given size at the given position. The handle
+    /// should contain a framebuffer of size width * height * 4 (aligned up). Its
```
Comment (Orycterope) [inline libuser/src/vi.rs]:
```suggestion
    /// should contain a framebuffer of size width * height * 4 (aligned up to page_size). Its
```

File: sm/src/main.rs [hunk 2]
```diff
@@ -1,7 +1,23 @@
+//! Service Manager
+//!
+//! Services are system processes running in the background which wait for
+//! incoming requests. When a process wants to communicate with a service, it
+//! first needs to get a handle to the named service, and then it can communicate
+//! with the service via inter-process communication (each service has a name up
+//! to 8 characters).
```
Comment (Orycterope) [inline sm/src/main.rs]:
```suggestion
//! to 7 characters).
```

File: sm/src/main.rs [hunk 3]
```diff
@@ -1,7 +1,23 @@
...
+//!
+//! Handles for services are retrieved from the service manager port, "sm:", and
+//! are released via svcCloseHandle or when a process is terminated or crashes.
+//! Manager service "sm:m" also exists. Services are an abstraction of ports,
```
Comment (Orycterope) [inline sm/src/main.rs]: same remark

File: libuser/src/types.rs [hunk 4]
```diff
@@ -27,30 +43,58 @@ impl Handle {
...
-        syscalls::close_handle(self.0.get());
+        let _ = syscalls::close_handle(self.0.get());
     }
 }
 
+/// A fake reference to a Handle. Has the same representation as a real Handle,
+/// but is bound to the real handle's lifetime.
+///
+/// This pattern allows for passing handle arrays without giving up ownership of
+/// the handle, and without an expensive conversion from an arrray of pointers to
```
[the remark this answers is not in the document]
Comment (roblabla) [reply libuser/src/types.rs]: I'm totally applying this suggestion

**Comment by roblabla** [comment]
Fixed.

**Pull request merged**